src/BrandwatchReactAuth.js

```diff
@@ -34,18 +36,28 @@ export default class BrandwatchReactAuth extends Component {
     this.handleGetToken()
       .then(token => token
         ? this.setState(() => ({ loggedIn: true }))
-        : new Promise((resolve, reject) => {
-          return backupDomain ? resolve(this.handleGetToken(backupDomain)) : reject()
-        })
-        .then(token => {
-          return token && backupRedirect ? window.location.replace(backupRedirect) : Promise.reject()
-        })
-        .catch(e => {
-          return window.location.replace(this.store.loginUrl)
-        })
+        : new Promise((resolve, reject) =>
+          backupDomain ? resolve(this.handleGetToken(backupDomain)) : reject()
+        )
+        .then(token =>
+          token && backupRedirect ? window.location.replace(backupRedirect) : Promise.reject()
+        )
+        .catch(() => window.location.replace(this.store.loginUrl))
     );
   }
 
+  handleError() {
+    const { backupDomain, backupRedirect } = this.props;
+    const { loginUrl } = this.store;
+
+    if (backupDomain && backupRedirect) {
+      return this.handleGetToken(backupDomain)
+        .then(token => window.location.replace(token ? backupRedirect : loginUrl));
+    }
+
+    return window.location.replace(this.store.loginUrl);
```

Comment:
Is this flow the same as what's happening inside the mount?

Reply:
almost, only this gets called regardless of the state of your token, the `mount` flow could be refactored to try and isolate the commonality.